src/components/DeviceItem/DeviceItem.tsx

```diff
@@ -0,0 +1,61 @@
+import { useMemo, useCallback } from "react";
+import { get, find } from "lodash";
+import { Link, Title, TwoProperties } from "@deskpro/app-sdk";
+import { useExternalLinks } from "../../hooks";
+import { getOS, getCapacity } from "../../utils";
+import { LansweeperLogo } from "../common";
+import type { FC, MouseEventHandler } from "react";
+import type { Maybe } from "../../types";
+import type { Site, Device } from "../../services/lansweeper/types";
+
+export type Props = {
+  device: Device;
+  sites: Site[];
+  siteId: Maybe<Site["id"]>;
+  onClickTitle?: () => void,
+};
+
+const DeviceItem: FC<Props> = ({ sites, siteId, device, onClickTitle }) => {
+  const { getDeviceLink } = useExternalLinks();
+  const site = useMemo(() => find(sites, { id: siteId }), [sites, siteId]) as Maybe<Site>;
+  const link = useMemo(() => {
+    return getDeviceLink(site?.name, device.key);
+  }, [site?.name, device.key, getDeviceLink]);
+  const onClick: MouseEventHandler<HTMLAnchorElement> = useCallback((e) => {
+    e.preventDefault();
+    onClickTitle && onClickTitle();
+  }, [onClickTitle]);
+  const os = useMemo(() => getOS(get(device, ["operatingSystem"])), [device]);
+  const capacity = useMemo(() => getCapacity(get(device, ["diskPartitions"])), [device]);
```

Comment:
`getCapacity` is lightweight to run so not sure it needs a memo?

Reply:
I use the following rule: if there is even one expression or any calculations, then the function is complex and I use `useMemo`.

Reply:
in your opinion, when a "memo" is needed and when not

Reply:
It is needed when an expensive operation can be cached. `getCapacity` did a couple of `if` statements. That doesn't need to be cached. Something like a SHA1 hash or a deeply nested tree, that might need to be cached, otherwise we're making code more complicated and using more memory than we need to.
source: https://react.dev/reference/react/useMemo#usage

Reply:
Got it. Thanks. Removed `useMemo` from `os` and `capacity`